### ISSUE TITLE
ObjectBoxLiveData fires twice

### ISSUE THREAD

**Issue by arnis71** [objectbox/objectbox-java#248]
ObjectBoxLiveData observer fires twice in a fragment when I reopen it. Regular LiveData works fine under the same conditions. Is that an intended behaviour and I am missing something out or is it a bug?

## Issue Basics
- ObjectBox version (are using the latest version?): 1.2.0
- Reproducibility:  always

### Description
This happens when I leave the fragment in a view pager and open it later. On first open there it fires only once.

### Code
ViewModel
```
class PortraitVM: ViewModel() {
    private val repo by injectEverywhere<PortraitRepo>()

    val faces = repo.queryFaces().toLiveData()

    fun faceToBubbleMapper() = Function<PortraitFace, PortraitFaceBubble> {
        repo.bubbleFromFace(it)
    }
}
```
Fragment snippet
```
override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        vm.faces.observe(this, Observer { faces ->
            faces?.let { onFacesReady(it) }
        })
    }
```

### Logs & stackstraces
There is a setValue() function in the stack trace when the second value comes in.

## Misc
Seems like ObjectBoxLiveData thinks that I change the query somewhere and the observer is triggered.

**Comment by arnis71** [comment]
After commenting postValue(data) in MyObjectBoxLiveData listener I am not receiving any values and when I uncomment it I receive 2.

**Comment by arnis71** [comment]
After further investigating the issue I've found out that the second value comes in because of the lifecycle state changing. So how can I overcome this issue?

**Comment by greenrobot-team** [comment]
Thanks for reporting. This seems to be caused by the LiveData returning an existing value when an observer becomes active. I currently solved this by only publishing an initial value if there is none:
```
protected void onActive() {
    // called when the LiveData object has an active observer
    subscription = query.subscribe().onlyChanges().observer(listener);
    if (getValue() == null) {
        query.publish();
    }
}
```
However, this now has the disadvantage that if a data change happens while no observer is active the LiveData returns its cached, outdated data.
-ut

**Comment by greenrobot-team** [comment]
Well, do you consider my suggested fix an improvement? Or is the introduced disadvantage an issue as well? -ut

**Comment by greenrobot** [comment]
@arnis71 Could you please check 1.2.1 if that fixes the issue for you?

**Comment by khatv911** [comment]
@greenrobot : This is still an issue in version 1.5.0

**Comment by greenrobot-team** [comment]
@JulRaad95 Please open a new issue with more details, preferably a small project to reproduce the issue. -ut
